hal/unix/UartUnix.cpp

```diff
@@ -28,7 +28,11 @@ namespace hal
 
     UartUnix::~UartUnix()
     {
-        running = false;
+        LockReadThen([this]
+            {
+                running = false;
+                receivedData = nullptr;
+            });
```

Comment:
Please just use the standard idiom

```suggestion
        {
            std::unique_lock lock(receivedDataMutex);
            running = false;
            receivedData = nullptr;
        }
```